src/write_gcm.c

```diff
@@ -2275,59 +2275,136 @@ static monitored_resource_t *monitored_resource_create_from_fields(
   // Changes va_list into label_t array
   va_start(ap, project_id);
   for (int i = 0; i < num_labels; i++) {
-    labels[i].key = va_arg(ap, const char*);
-    labels[i].value = va_arg(ap, const char*);
+    labels[i].key = va_arg(ap, char*);
+    labels[i].value = va_arg(ap, char*);
   }
   va_end(ap);
   return monitored_resource_create_from_array(type, project_id, labels,
                                               num_labels);
 }
 
+typedef struct {
+  _Bool processing_labels;
+  char *label_name;
+  char *type;
```

Comment:
Let's group all of the label-related fields together. I'd suggest moving `type` to the top of the struct.